src/devices/src/virtio/net/event_handler.rs

```diff
@@ -8,15 +8,91 @@ use polly::event_manager::{EventManager, Subscriber};
 use utils::epoll::{EpollEvent, EventSet};
 
 use crate::virtio::net::device::Net;
-use crate::virtio::{VirtioDevice, RX_INDEX, TX_INDEX};
+use crate::virtio::{DeviceState, RX_INDEX, TX_INDEX};
 
-impl Subscriber for Net {
-    fn process(&mut self, event: &EpollEvent, _: &mut EventManager) {
-        if !self.is_activated() {
-            warn!("The device is not yet activated. Events can not be handled.");
-            return;
-        }
+impl Net {
+    fn process_activate_event(&self, event_manager: &mut EventManager) {
+        // The subscriber must exist as we previously registered activate_evt via
+        // `interest_list()`.
+        let self_subscriber = event_manager
+            .subscriber(self.activate_evt.as_raw_fd())
+            .unwrap();
+
+        event_manager
+            .register(
+                self.queue_evts[RX_INDEX].as_raw_fd(),
+                EpollEvent::new(EventSet::IN, self.queue_evts[RX_INDEX].as_raw_fd() as u64),
+                self_subscriber.clone(),
+            )
+            .unwrap_or_else(|e| {
+                error!(
```

Comment:
Do we want to continue the execution if we cannot register the events needed for this device? It might lead to silent failures.

Reply:
I believe that once a guest is booted (customer workload has possibly started) we should always avoid crashing.

A malfunctioning device has the blast-radius limited to operations on that device. The guest workload might or might not be affected and data might be recoverable if the malfunction is detected.